internal/kubernetes/operator/deployment/deployment.go

```diff
@@ -354,54 +350,92 @@ func buildReplicationSpec(atlasRepSpec []*mongodbatlas.AdvancedReplicationSpec)
 
 		replicationSpec.RegionConfigs = make([]*atlasV1.AdvancedRegionConfig, 0, len(replicationSpec.RegionConfigs))
 		for _, rc := range rs.RegionConfigs {
-			if rc == nil {
-				continue
+			var configAnalyticsSpecs *atlasv2.DedicatedHardwareSpec
+			var configElectableSpecs *atlasv2.HardwareSpec
+			var configReadOnlySpecs *atlasv2.DedicatedHardwareSpec
+			var configAutoScaling *atlasv2.AutoScalingV15
+			var priority *int
+			var providerName string
+			var regionName string
+			var backingProviderName string
+
+			switch {
+			case rc.AWSRegionConfig != nil:
```

Comment:
Same here. I feel like your experience would be much better if we merge those into single object